src/pages/GivePage/index.tsx

```diff
@@ -17,24 +21,40 @@ import { IEpoch, IMyUser } from '../../types';
 import { Box, Button, Flex, Modal, Panel, Text, Link } from '../../ui';
 import { SingleColumnLayout } from '../../ui/layouts';
 import { getPendingGiftsFrom } from '../AllocationPage/queries';
+import { FormInputField } from 'components';
 import { SaveState, SavingIndicator } from 'ui/SavingIndicator';
 
 import { EpochStatementDrawer } from './EpochStatementDrawer';
 import { GiveDrawer } from './GiveDrawer';
 import { GiveRow } from './GiveRow';
 import { MyGiveRow } from './MyGiveRow';
-import { getMembersWithContributions, PotentialTeammate } from './queries';
+import {
+  getCircleAllocationText,
+  getMembersWithContributions,
+  PotentialTeammate,
+  QUERY_KEY_CIRCLE_ALLOCATION_TEXT,
+} from './queries';
 
 export type Gift = Awaited<ReturnType<typeof getPendingGiftsFrom>>[number];
 
 // the amount of time to wait for inactivity to save
+const schema = z.object({
+  alloc_text: z
+
+    .string()
+    .max(500)
+    .refine(val => val.trim().length >= 40, {
+      message: 'Please write at least 40 characters.',
+    }),
+});
+type allocationTextSchema = z.infer<typeof schema>;
 const SAVE_BUFFER_PERIOD = 1000;
 
 const GivePage = () => {
   const address = useConnectedAddress();
   const {
     circle: selectedCircle,
-    myUser,
+    myUser: me,
```

Comment:
is this change necessary? you can use myUser directly to reduce changes.
you will reduce the changes alot if you kept myUser

Reply:
done